### PR TITLE
[WIP] Add in a script tuner to NUsight2

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

File: src/client/components/script_tuner/viewer/view_model.ts [hunk 1]
```diff
@@ -1,22 +1,285 @@
...
-    return RobotViewModel.of(this.model.currentRobotModel)
+  get name() {
+    return this.model.robot.name
+  }
+
+  @computed
+  get rightShoulderPitch() {
+    const servo = this.model.servos[0]
+
+    if (servo.frames.length) {
```
[the remark this answers is not in the document]
Comment (JosephusPaye) [reply src/client/components/script_tuner/viewer/view_model.ts]: Done

File: src/client/components/script_tuner/editor/line/controller.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    this.model = model
+  }
+
+  static of(model: Servo) {
+    return new LineEditorController(model)
+  }
+
+  @action
+  addFrame = (data: { time: number, angle: number }) => {
+    this.model.frames.push({
```
[the remark this answers is not in the document]
Comment (JosephusPaye) [reply src/client/components/script_tuner/editor/line/controller.ts]: Done

File: src/client/components/script_tuner/editor/line/view.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    }, { once: true })
+  }
+
+  private endDrag() {
+    this.isDragging = false
+    this.selectedElement = undefined
+  }
+
+  private onMouseDown = ({ nativeEvent: event }: React.MouseEvent, viewModel: LineEditorViewModel) => {
+    if ((event.target as HTMLElement).dataset.draggable) {
```
[the remark this answers is not in the document]
Comment (JosephusPaye) [reply src/client/components/script_tuner/editor/line/view.tsx]: Done

**Comment by TrentHouliston** [comment]
Closing so @JosephusPaye can reopen

**Comment by JosephusPaye** [comment]
Superseded by #234.

**Pull request closed without merging**